test/fixtures/parser/postgres/postgres_create_type.sql

```diff
@@ -0,0 +1,6 @@
+-- https://www.postgresql.org/docs/current/sql-createtype.html
+CREATE TYPE foo;

```

Comment:
Since this PR adds to the ANSI grammar, would it make sense for this test to be in the ANSI directory rather than Postgres?

Reply:
I thought about this, and in the end - given it's explicitly a postgres query it made sense to test there. Eventually the code in the ansi dialect may get more generic (e.g. with Sql server?) and then it might be two test cases, served by the same code. In that instance I think it makes sense to keep the test cases dialect specific.